File: app/src/main/java/eu/secrit/deaftalk/Eintrag.java
package eu.secrit.deaftalk;

public class Eintrag {

    //Eintrags Elemente
    private String eintrag;
    private boolean markiert;
    private boolean fromMeOrYou;

    public String getEintrag(){ return eintrag;}
    public void setEintrag(String eintrag){ this.eintrag = eintrag; }
    public boolean isMarkiert() { return markiert; }
    public void setMarkiert(boolean markiert) { this.markiert = markiert;}
    public boolean isFromMeOrYou() { return fromMeOrYou; }
    public void setFromMeOrYou(boolean fromMeOrYou) { this.fromMeOrYou = fromMeOrYou; }

    //Konstruktoren
    //Wird aufgerufen, sobald die Klasse mit "new" initialisiert wird
    public Eintrag(){
        this.setEintrag("Hier ist nichts drin");
        this.setMarkiert(false);
        this.setFromMeOrYou(true);
    }

    public String toString(){
        String retMarkiert = "";
        if (this.isMarkiert() == true){ retMarkiert = "X";}
        String retFromMeOrYou = "Gast";
        if (this.isFromMeOrYou() == true) {retFromMeOrYou = "Ich"; }

        return  retMarkiert + " " + retFromMeOrYou + ": "+ getEintrag();
    }
}

File: lib/models/eintrag.dart
/// Ein einzelner Chat-Eintrag (entspricht der Java-Klasse `Eintrag`).
class Eintrag {
  String eintrag; // der Nachrichtentext
  bool markiert; // ausgewählt (im Original mit "X" markiert)
  bool fromMeOrYou; // true = Ich, false = Gast

  Eintrag({
    this.eintrag = 'Hier ist nichts drin',
    this.markiert = false,
    this.fromMeOrYou = true,
  });

  /// Anzeige-String wie im Original: "[X] Ich/Gast: text"
  @override
  String toString() {
    final mark = markiert ? 'X ' : '';
    final wer = fromMeOrYou ? 'Ich' : 'Gast';
    return '$mark$wer: $eintrag';
  }

  Map<String, dynamic> toJson() => {
        'eintrag': eintrag,
        'fromMeOrYou': fromMeOrYou,
      };

  factory Eintrag.fromJson(Map<String, dynamic> json) => Eintrag(
        eintrag: json['eintrag'] as String? ?? '',
        fromMeOrYou: json['fromMeOrYou'] as bool? ?? true,
        markiert: false,
      );
}
